File: Grind75/ValidParentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stk = []
        for e in s:
            if e == "(":
                stk.append(")")
            elif e == "[":
                stk.append("]")
            elif e == "{":
                stk.append("}")
            elif not stk or stk.pop() != e:
                return False
        return not stk
        

class Solution:
    def isValid(self, s: str) -> bool:
        stk = []
        dict = {")": "(", "}":"{", "]":"["}
        for e in s:
            if e not in dict:
                stk.append(e)
            elif not stk or dict[e] != stk.pop():
                return False
        return not stk

File: Grind75/ValidParentheses.java
package Grind75;

import java.util.Stack;

public class ValidParentheses {
    public static void main(String[] args) {
        ValidParentheses solution = new ValidParentheses();
        String s = "{}[]()";
        System.out.println(solution.isValid(s));
    }
    public boolean isValid(String s){
        Stack<Character> stack = new Stack<Character>();

        for(char c : s.toCharArray()){
            if(c == '('){
                stack.push(')');
            }
            else if(c == '{'){
                stack.push('}');
            }
            else if(c == '['){
                stack.push(']');
            }
            else if(stack.isEmpty() || stack.pop() != c){
                return false;
            }
        }
        return stack.isEmpty();
    }
}

